File: libs/typeorm/src/entities/events/EventOperationFilter.ts
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';

// eslint-disable-next-line import/no-cycle
import { EventOperation } from './EventOperation';

export enum FilterType {
  EQUALS = 'EQUALS',
  NOT_EQUALS = 'NOT_EQUALS',
  CONTAINS = 'CONTAINS',
  NOT_CONTAINS = 'NOT_CONTAINS',
  STARTS_WITH = 'STARTS_WITH',
  ENDS_WITH = 'ENDS_WITH',
  GREATER_THAN = 'GREATER_THAN',
  LESS_THAN = 'LESS_THAN',
  GREATER_THAN_OR_EQUALS = 'GREATER_THAN_OR_EQUALS',
  LESS_THAN_OR_EQUALS = 'LESS_THAN_OR_EQUALS',
  IS_EMPTY = 'IS_EMPTY',
  IS_NOT_EMPTY = 'IS_NOT_EMPTY',
}

@Entity('channels_events_operations_filters')
export class EventOperationFilter {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => EventOperation, _ => _.filters, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'operationId' })
  operation?: EventOperation;

  @Column('uuid')
  operationId: string;

  @Column('simple-enum', { enum: FilterType })
  type: FilterType;

  @Column('text')
  left: string;

  @Column('text')
  right: string;
}

File: libs/gomodels/event_operation_filter.go
package model

import (
	"github.com/satont/twir/libs/types/types/events"
)

type EventOperationFilter struct {
	ID          string                          `gorm:"primaryKey;column:id;type:UUID;default:uuid_generate_v4()" json:"id"`
	OperationID string                          `gorm:"column:operationId;type:UUID;" json:"operationId"`
	Type        events.EventOperationFilterType `gorm:"column:type;type:TEXT;" json:"type"`
	Left        string                          `gorm:"column:left;type:TEXT;" json:"left"`
	Right       string                          `gorm:"column:right;type:TEXT;" json:"right"`

	Operation *EventOperation `gorm:"foreignkey:OperationID" json:"operation"`
}

func (c *EventOperationFilter) TableName() string {
	return "channels_events_operations_filters"
}
